test/dropRepeatsBy.js

```diff
@@ -0,0 +1,35 @@
+var R = require('../source/index.js');
+var eq = require('./shared/eq.js');
+
+
+describe('dropRepeatsBy', function() {
+  var objs = [
+    {i: 1}, {i: 2}, {i: 3}, {i: -4}, {i: 5}, {i: 3}
+  ];
+  var objs2 = [
+    {i: 1}, {i: -1}, {i: 1}, {i: 2}, {i: 3},
+    {i: 3}, {i: -4}, {i: 4}, {i: 5}, {i: 3}
+  ];
+  var fn = ({ i, n, ...rest }) => ({ i : Math.abs(i), ...rest });
+
+  it('removes repeated elements based on predicate', function() {
+    eq(R.dropRepeatsBy(fn, objs2), objs);
+    eq(R.dropRepeatsBy(fn, objs), objs);
```

Comment:
Do you think there could be a risk for a false positive if either `dropRepeatsBy` or `fn` mutates `objs`? In general I prefer keeping tests isolated to avoid collisions. Do you think test data could simply be inlined in each test?

Reply:
if `dropRepeatsBy` itself mutates the data, we've got serious problems.

But I agree that tests should test one thing, which usually means a single assertion, or perhaps a before and after assertion.

These two should be in separate tests.  Good catch.,

Reply:
`dropRepeatsBy` should never mutate the data. Why would it?

Reply:
Btw the point on test isolation is valid. This test file is based on [https://github.com/ivanfilhoz/ramda/blob/feature/dropRepeatsBy/test/dropRepeatsWith.js](test/dropRepeatsWith.js). Should we fix it as well?

Reply:
yes, we probably should.  We haven't been as consistent with test code as with the production code.  This is a minor nit, but yes, let's fix it in both places.

Reply:
Okay, done. Please check new version. I think most of the risks have been properly addressed.

Reply:
> `dropRepeatsBy` should never mutate the data. Why would it?

Sorry for the confusion here. Of course it should not. I was just thinking about the case where it accidentally did. We sometimes miss cases of mutation during development or code review, so tests can be written in such a way to detect them. I was only asking whether such test could detect this problem as I wasn't convinced it was.

Thanks for the changes.

Reply:
I don't think my comment has been addressed but I'm happy to revisit some of these tests myself separately. So will resolve this conversation for now.